### DISCUSSION TITLE
USB host emulation on ESP32

### DISCUSSION THREAD

**Discussion by marcel-licence** [marcel-licence/esp32_usb_midi#3] [Ideas]
In a youtube video comment I've received a link to an interesting project:
- https://github.com/sdima1357/esp32_usb_soft_host
- https://hackaday.com/2021/03/26/usb-comes-to-the-esp32/
It seems to be possible to do some USB stuff. But actually the project is not compatible with arduino yet and there is no MIDI implementation yet (and the stuff in between).
Do you have any ideas if that could be a feasible solution?

**Comment by 01GOD** [comment]
Hi! Been working on that today. Supposedly an S3, S2, and C3 thing. 
Found this for USB host mode so an ESP32-S3 supposedly can receive MIDI, but it wasn't functioning when tried it earlier today: 
https://github.com/touchgadget/esp32-usb-host-demos/

And this discussion has some useful info: https://github.com/touchgadget/esp32-usb-host-demos/issues/1

Really should be easy to get functioning, so hopefully somebody in the community will figure it out soon. Will be nice to have it in that ESP32 Basic Synth. There is (confirmed functioning) for USB OTG mode to send MIDI from ESP32-S3 through MIDI: https://github.com/esp32beans/ESP32USBMIDI
